bundle/src/files.rs

```diff
@@ -172,22 +175,77 @@ pub struct FileSet {
     pub file_set_type: FileSetType,
     pub files: Vec<BundledFile>,
     pub glob: String,
+    #[serde(flatten)]
+    pub test_runner_report: Option<FileSetTestRunnerReport>,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "pyo3", gen_stub_pyclass, pyclass(get_all))]
+#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+pub struct FileSetTestRunnerReport {
     /// Added in v0.6.11. Populated when parsing from BEP, not from junit globs
-    pub resolved_status: Option<JunitReportStatus>,
+    pub resolved_status: TestRunnerReportStatus,
+    /// Added in v0.9.3. Populated when parsing from BEP, not from junit globs
+    #[cfg_attr(feature = "wasm", tsify(type = "number"))]
+    #[serde(default, with = "ts_milliseconds")]
+    pub resolved_start_time_epoch_ms: DateTime<Utc>,
+    /// Added in v0.9.3. Populated when parsing from BEP, not from junit globs
+    #[cfg_attr(feature = "wasm", tsify(type = "number"))]
+    #[serde(default, with = "ts_milliseconds")]
+    pub resolved_end_time_epoch_ms: DateTime<Utc>,
+}
+
+#[cfg(feature = "pyo3")]
+#[gen_stub_pymethods]
+#[pymethods]
+impl FileSetTestRunnerReport {
+    #[cfg(feature = "pyo3")]
+    #[new]
+    pub fn new(
+        resolved_status: TestRunnerReportStatus,
+        resolved_start_time_epoch_ms: DateTime<Utc>,
+        resolved_end_time_epoch_ms: DateTime<Utc>,
+    ) -> Self {
+        Self {
+            resolved_status,
+            resolved_start_time_epoch_ms,
+            resolved_end_time_epoch_ms,
+        }
+    }
+}
+
+impl From<TestRunnerReport> for FileSetTestRunnerReport {
+    fn from(test_runner_report: TestRunnerReport) -> Self {
+        Self {
+            resolved_status: test_runner_report.status,
+            resolved_start_time_epoch_ms: test_runner_report.start_time,
+            resolved_end_time_epoch_ms: test_runner_report.end_time,
+        }
+    }
+}
+
+impl From<FileSetTestRunnerReport> for TestRunnerReport {
+    fn from(test_runner_report: FileSetTestRunnerReport) -> Self {
+        Self {
+            status: test_runner_report.resolved_status,
+            start_time: test_runner_report.resolved_start_time_epoch_ms,
+            end_time: test_runner_report.resolved_end_time_epoch_ms,
+        }
+    }
 }
```

Comment:
Interop for `meta.json` and `context/junit` types. The split isn't just for field naming, but for (de)serialization implementation